lib/data_source/client/github.rb

```diff
@@ -29,30 +42,86 @@ def user_repos
           direction: 'desc',
           per_page: 100
         }
-        with_cache('user_repos') do
-          request('/user/repos', params)
+        responses = with_cache(:user_repos) do
+          all_user_repos = merge_multipage_results('/user/repos', params, 100)
+          all_user_repos
+        end
+
+        all_user_repos = Array.new
+        responses.each do |response|
+          all_user_repos = all_user_repos + response
         end
+        all_user_repos
```

Comment:
I think this is iterating unnecessarily. You can return `responses` directly. And even remove it all together since `with_cache` is the last call in the method (But also remove the assignation of the variable if it's not used (`responses = ...`))

```suggestion
```

Reply:
actually no, because responses is an array of response, each element is a response for a page.

The iterating here is to concatenate the elements inside each response together.

Reply:
But it's just copying them over another array isn't it? `all_user_repos` is also an array, right?

Reply:
It’s more like flatting:
From [[a],[b],[c]] to [a, b ,c]

Reply:
I see. In that case you can use Ruby's `Array.flat_map` method:

```ruby
responses.flat_map
```

Reply:
yep, I have updated this in my previous commit. this comment is outdated.